COLLABORATOR_GUIDE.md

```diff
@@ -111,3 +118,112 @@ layout: contribute.hbs
+
+- All React Components should be placed within the `components` folder.
+- Each Component should be placed whenever possible within a sub-folder, which we call the "Domain" of the Component
+  - The domain is the representation of where these Components belong to or where will be used.
+  - For example, Components used within Article Pages or that are part of the structure of an Article or the Article Layouts, should be placed within `components/Article`
+- Each component should have its own folder with the name of the Component
+- The structure of each component folder follows the following template:
+  ```text
+  - ComponentName
+    - index.tsx // the component itself
+    - index.module.scss // all styles of the component are placed there
+    - index.stories.tsx // component Storybook stories
+    - __tests__ // component tests (such as unit tests, etc)
+      - index.test.tsx
+  ```
+- React Hooks belonging to a single Component should be placed within the Component's folder
+  - If the Hook as a wider usability or can be used by other Components, then it should be placed at the root `hooks` folder.
+- If the Component has "sub-components" they should follow the same philosophy as the Component itself.
+  - For example, if the Component `ComponentName` has a sub-component called `SubComponentName`, then it should be placed within `ComponentName/SubComponentName`
+
+#### How a new Component should look like when freshly created
+
+```tsx
+import styles from './index.module.scss';
+import type { FC } from 'react';
+
+type MyComponentProps = {}; // The types of the Props of your Component
+
+const MyComponent: FC<MyComponentProps> = ({ prop1, prop2... }) => (
+  // Actual code of my Component
+);
+
+export default MyComponent;
+```
+
+### Best practices for Component development in general
+
+- Only spread props `{ ... }` on the definition of the Component (Avoid having a variable named `props`)
+- Avoid importing `React`, only import the modules from React that you need
+- When importing types use `import type { NameOfImport } from 'module'`
+- When defining a Component use the `FC` type from React to define the type of the Component
+  - When using `children` as a prop, use the `FC<PropsWithChildren<MyComponentProps>>` type instead
+  - Alterenatively you can define your type as `type MyComponentProps = PropsWithChildren<{ my other props}>`
+- Each Props type should be prefixed by the name of the Component
+- Components should always be the `default` export of a React Component file
+- Avoid using DOM/Web APIs/`document`/`window` API access within a React Component. Use utilities or Hooks when you need a Reactive state
+- Avoid making your Component too big. Deconstruct it into smaller Components/Hooks whenever possible
+
+## Unit Tests and Storybooks
+
+Each new feature or bug fix should be accompanied by a unit test (when deemed valuable). We use [Jest][] as our test runner and [React Testing Library][] for our React unit tests.
```

Comment:
Some more reference style links here that need to be wired up

Reply:
same as above - thanks once more @mikeesto